crates/core/src/db/queries/metadatas.rs

```diff
@@ -264,23 +267,36 @@ pub fn list(
         );
     }
 
-    if let Some(offerers) = offerers {
+    // ignoring the with_offers=false case for now, since it is more complicated and we dont currently have a need
+    let should_query_bid_receipts =
+        offerers.is_some() || (with_offers.is_some() && with_offers.unwrap_or(false));
+
+    if should_query_bid_receipts {
+        let mut bid_receipts_conditions = Condition::all().add(
+            Expr::tbl(BidReceipts::Table, BidReceipts::Metadata)
+                .equals(Metadatas::Table, Metadatas::Address),
+        );
+
+        if let Some(offerers) = offerers {
+            bid_receipts_conditions = bid_receipts_conditions
+                .add(Expr::col((BidReceipts::Table, BidReceipts::Buyer)).is_in(offerers))
+                .add(Expr::tbl(BidReceipts::Table, BidReceipts::PurchaseReceipt).is_null())
+                .add(Expr::tbl(BidReceipts::Table, BidReceipts::CanceledAt).is_null());
+        } else if let Some(with_offers) = with_offers {
+            if with_offers {
+                bid_receipts_conditions = bid_receipts_conditions
+                    .add(Expr::tbl(BidReceipts::Table, BidReceipts::PurchaseReceipt).is_null())
+                    .add(Expr::tbl(BidReceipts::Table, BidReceipts::CanceledAt).is_null());
+            }
+        }
```

Comment:
Instead of doing special logic for one or the other with_offers should tack on filters to bid receipt where canceled_at isn't null but keep the offerers logic. If someone calls the query with_offers and offerers then they should get the nfts for the addresses set as offerers. If they just set with_offers should pick any nfts with an offer for the supplied auction_houses.

Reply:
Updated to remove the `else`, which I think is what you were asking for. 

Also, this PR is still a draft because I am getting a runtime error  I dont know how to track down:

> Caused by:
>     Column `update_authority_address` was not present in query

Reply:
The reported bug is fixed in this PR